render/topologies_test.go

```diff
@@ -412,17 +412,17 @@ func TestContainerRenderer(t *testing.T) {
 			LabelMinor: serverHostName,
 			Rank:       serverContainerImageID,
 			Pseudo:     false,
-			Adjacency:  report.MakeIDList(clientContainerID, render.UncontainedID, render.TheInternetID),
+			Adjacency:  report.MakeIDList(clientContainerID, render.TheInternetID),
 			Origins:    report.MakeIDList(serverContainerNodeID, server80NodeID, serverProcessNodeID, serverHostNodeID),
 			AggregateMetadata: render.AggregateMetadata{
 				render.KeyBytesIngress: 150,
 				render.KeyBytesEgress:  1500,
 			},
 		},
-		render.UncontainedID: {
-			ID:                render.UncontainedID,
+		fmt.Sprintf("%s:%s", render.UncontainedID, serverHostName): {
+			ID:                fmt.Sprintf("%s:%s", render.UncontainedID, serverHostName),
```

Comment:
These aren't report-level node ids; these are for rendered nodes. None of
the functions in this file use constructors for id. I agree it would be a
good thing, but I think it's beyond the scope of this PR.

What do you think? Do it this PR or the next?

On Friday, 19 June 2015, Peter Bourgon <notifications@github.com> wrote:

> In render/topologies_test.go
> <https://github.com/weaveworks/scope/pull/264#discussion_r32848666>:
>
> >  			Origins:    report.MakeIDList(serverContainerNodeID, server80NodeID, serverProcessNodeID, serverHostNodeID),
> >  			AggregateMetadata: render.AggregateMetadata{
> >  				render.KeyBytesIngress: 150,
> >  				render.KeyBytesEgress:  1500,
> >  			},
> >  		},
> > -		render.UncontainedID: {
> > -			ID:                render.UncontainedID,
> > +		fmt.Sprintf("%s:%s", render.UncontainedID, serverHostName): {
> > +			ID:                fmt.Sprintf("%s:%s", render.UncontainedID, serverHostName),
>
> Please use an ID constructor.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/weaveworks/scope/pull/264/files#r32848666>.
>